crates/solidity/outputs/cargo/crate/src/lib.rs

```diff
@@ -1,7 +1,5 @@
-mod legacy;
-
-pub mod language;
-pub mod syntax;
+mod generated;
+pub use generated::*;
```

Comment:
Perhaps related to the earlier comment, but we had this separation (public API vs generated API) because it was much easier to specify and annotate the public API (`language` and `syntax` modules) directly here, re-exporting the types we needed from the single `generated` module. I wonder why were they removed?

Reply:
I wasn't convinced about the previous structure i.e. `syntax/nodes` and `language`. I would like us to think about this separately.

Reply:
Sure. Happy to discuss updating these namespaces, but not sure if that is related to my question.
I'm talking about the separation between:

- codegen writing the files/templates to `src/generated/*`
- hand-authored public API files in `src/XXX` that re-exports types from `generated::*`

We use this separation for two reasons:

1. It is much harder to enforce a clean public API from codegen.
2. We want to add/augment the above types with hand-authored APIs.

Files like `src/language/mod.rs` and `src/syntax/nodes/mod.rs` make it much easier to do that, by simply re-exporting the bits they want to expose to users, and keeping `generated:**` private. This will scale much better with the addition of more types/APIs later, for example in the AST.

Same exact thing for the TypeScript API files like `package/src/language/index.ts` and `package/src/syntax/nodes/index.ts`, re-exporting from the NAPI-generated `package/src/generated/index.js`